tools/cli/app_test.go

```diff
@@ -142,19 +144,29 @@ func (s *cliAppSuite) TearDownTest() {
 	s.mockCtrl.Finish() // assert mock’s expectations
 }
 
-func (s *cliAppSuite) testcaseHelper(testcases []testcase) {
-	for _, tt := range testcases {
-		s.T().Run(tt.name, func(t *testing.T) {
-			if tt.mock != nil {
-				tt.mock()
-			}
-			err := s.app.Run(strings.Split(tt.command, " "))
-			if tt.err != "" {
-				assert.ErrorContains(t, err, tt.err)
-			} else {
-				assert.NoError(t, err)
-			}
-		})
+func (s *cliAppSuite) SetupSubTest() {
+	// this is required to re-establish mocks on subtest-level
+	// otherwise you will confusingly see expectations from previous test-cases on errors
```

Comment:
urgh... I really don't like the suite method of testing :/ 

ty for the fix though, really annoying bug to fix in tests.

Reply:
Agree. I think it counters with Go idea that everything is explicit and there is no "magic". Suite definitely brings some. Unfortunately, there are so many tests that migrating seems too much of work.
...which reminds me, though - maybe I should just use ChatGPT for that!